### ISSUE TITLE
Bug: Indented YAML Frontmatter does not retain indention

### ISSUE THREAD

**Issue by andy-bond** [edheltzel/nunjucks-for-visual-studio-code#2]
Hi, first of all, just wanted to say that I think this extension works really well most of the time and I hope you continue to improve it :)

## Bug

With only the Better Nunjucks extension enabled, if I have frontmatter in my file (.html) that has indented properties, formatting the document will remove the indention.

This is pretty critical for [11ty Pagination](https://www.11ty.dev/docs/pagination/), which is how I noticed the problem.

## Gif
![better-nunjucks-format](https://user-images.githubusercontent.com/10934345/164896778-b51771d2-ecb1-4db7-b31f-8c180e534ff4.gif)

**Comment by edheltzel** [comment]
oh, that's interesting...

@andy-bond – Just curious if you set the grammar to `nunjucks` or is this still `html`?? 

I'm assuming you've got auto-formatting for `html` turned on in your workspace or global settings.  If that is the case, I'd say this is more of an issue with the formatting for `html`, than an bug with the extension. There are several issues for vscode back to 2017 that point out the html formating breaks front-matter (I noted a workaround for my jekyll snippets extension in this issue) - https://github.com/microsoft/vscode/issues/24088

So two solutions, both not super ideal but definitely get you pointed in the write direction:

1. The fastest workaround would be to disable auto-formatting or format on save by adding the following to your global or workspace settings. If you want to format HTML you'll have to select the code block and they choose `Format Selection` from the command prompt or by right-clicking
```json
  //Formatting
  "editor.formatOnSave": true,
  "[html]": {
    "editor.formatOnSave": false
  },
```

2. If you use or install the [Prettier Extension](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) and set it as your default formatter for html that should resolve your issue.
```json
 //Formatting
  "editor.formatOnSave": true,
  "[html]": {
    "editor.formatOnSave": true,
    "editor.defaultFormatter": "esbenp.prettier-vscode"
  },
```

Let me know if that helps... 🙂

**Comment by andy-bond** [comment]
You are correct - enabling Prettier and setting it as the default resolved this issue - thanks!

While I have you... Now that that is resolved what can I do to prevent Nunjucks syntax from being collapsed into one line rather than retaining the "blocks" I have created? 

Gif below:
![better-nunjucks-format-collapse](https://user-images.githubusercontent.com/10934345/164912401-1693d9ec-4fda-4db0-b333-9c3e0ba30a71.gif)

**Comment by edheltzel** [comment]
> If I set the grammar to Nunjucks and then try to format, it says I don't have a formatter installed for Nunjucks :/
Try this one out - [Nunjucks Template Formatter](https://marketplace.visualstudio.com/items?itemName=okitavera.vscode-nunjucks-formatter)

> While I have you... Now that that is resolved what can I do to prevent Nunjucks syntax from being collapsed into one line rather than retaining the "blocks" I have created?

It's kind of hard to tell what you're looking to do from the gif - if you're talking about code collapsing you can try this out from the [README](https://github.com/edheltzel/nunjucks-for-visual-studio-code#known-issues) Just make sure you're indenting and adding line breaks between tags. 

If you're **not** wanting the white space inside of the generated `html` you can always use the [Whitespace Control syntax](https://mozilla.github.io/nunjucks/templating.html#whitespace-control)

**Comment by andy-bond** [comment]
Thought I'd mention that I tried enabling both your extension and the Nunjucks Template Formatter and that result turned out pretty good in terms of dealing with Nunjucks.

The only reason I hesitate to use the Nunjucks Template Formatter is that it sets all HTML files (no nunjucks) to "HTML Nunjucks". I am in a monorepo with other projects, so it is worrying to see "HTML Nunjucks" at the bottom of VS Code and all of the HTML files having a Nunjucks icon.
